modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinMultiplatformClientCodegen.java

```diff
@@ -0,0 +1,618 @@
+        supportingFiles.add(new SupportingFile(infraSrc + "OctetByteArray.kt.mustache", infraDest, "OctetByteArray.kt"));
+        if (dateLibrary == Options.DateLibrary.KOTLINX) {
+            supportingFiles.add(new SupportingFile(infraSrc + "DateTime.kt.mustache", infraDest, "DateTime.kt"));
+        }
+
+        // Auth
+        final String authDest = srcDir + File.separator + "auth";
+        final String authSrc = "common/main/auth/";
+        supportingFiles.add(new SupportingFile(authSrc + "ApiKeyAuth.kt.mustache", authDest, "ApiKeyAuth.kt"));
+        supportingFiles.add(new SupportingFile(authSrc + "Authentication.kt.mustache", authDest, "Authentication.kt"));
+        supportingFiles.add(new SupportingFile(authSrc + "HttpBasicAuth.kt.mustache", authDest, "HttpBasicAuth.kt"));
+        supportingFiles.add(new SupportingFile(authSrc + "HttpBearerAuth.kt.mustache", authDest, "HttpBearerAuth.kt"));
+        supportingFiles.add(new SupportingFile(authSrc + "OAuth.kt.mustache", authDest, "OAuth.kt"));
+
+        // Multiplatform default includes
+        defaultIncludes.add("io.ktor.client.request.forms.InputProvider");
+        defaultIncludes.add("io.ktor.client.statement.HttpResponse");
+        defaultIncludes.add(packageName + ".infrastructure.Base64ByteArray");
+        defaultIncludes.add(packageName + ".infrastructure.OctetByteArray");
+
+        // Multiplatform type mapping
+        typeMapping.put("number", "kotlin.Double");
+        typeMapping.put("file", "OctetByteArray");
+        typeMapping.put("binary", "OctetByteArray");
+        typeMapping.put("ByteArray", "Base64ByteArray");
+        typeMapping.put("object", "kotlinx.serialization.json.JsonObject");
+
+        // Multiplatform import mapping
+        importMapping.put("BigDecimal", "kotlin.Double");
+        importMapping.put("String", "kotlin.String");
+        importMapping.put("UUID", "kotlin.String");
+        importMapping.put("URI", "kotlin.String");
+        importMapping.put("InputProvider", "io.ktor.client.request.forms.InputProvider");
+        importMapping.put("File", "io.ktor.client.statement.HttpResponse");
+        importMapping.put("Base64ByteArray", packageName + ".infrastructure.Base64ByteArray");
+        importMapping.put("OctetByteArray", packageName + ".infrastructure.OctetByteArray");
+
+        switch (dateLibrary) {
+            case STRING: {
+                typeMapping.put("date-time", "String");
+                typeMapping.put("date", "String");
+                typeMapping.put("Date", "String");
+                typeMapping.put("DateTime", "String");
+                importMapping.put("Timestamp", "kotlin.String");
+                importMapping.put("LocalDateTime", "kotlin.String");
+                importMapping.put("LocalDate", "kotlin.String");
+                importMapping.put("LocalTime", "kotlin.String");
+            }
+            case KOTLINX: {
+                typeMapping.put("date-time", "LocalDateTime");
```

Comment:
Mapping the `date-time` field to `kotlinx.datetime.LocalDateTime` may not be appropriate.

[`date-time` corresponds to RFC3339](http://spec.openapis.org/oas/v3.0.3#data-types) which includes the time zone offset, e.g. `2020-07-31T07:16:15Z`, `2020-07-31T09:16:15+02:00`.

`kotlinx.datetime.LocalDateTime` does not include any time zone offset, and in fact `LocalDateTime.parse(..)` fails to parse both of the above examples.

See a similar issue and discussion which changed [`java.time.LocalDateTime` to `java.time.OffsetDateTime`](https://github.com/swagger-api/swagger-codegen/pull/3015).

Unfortunately, `kotlinx.datetime` does not currently have an analogue for `java.time.OffsetDateTime`. `kotlinx.datetime.Instant` currently maps to / from only the first example time above. One potential approach that comes to mind is to map to a custom `OffsetDateTime` type until `kotlinx.serialization` adds support for common formats such as ISO 8601 / RFC 3339.

```kotlin
class OffsetDateTime(
  val localDateTime: kotlinx.serialization.LocalDateTime,
  val offset: kotlinx.serialization.ZoneOffset,
)
```

Reply:
That is indeed [correct](https://github.com/Kotlin/kotlinx-datetime#type-use-cases)

Reply:
But maybe an instant would be more suitable here?

Reply:
Yeah something like a custom offset would probably be best, but I guess that means I have write some parsing...

Reply:
I hadn't checked this at all (just copied it over and changed it from Java to kotlinx), so feel free to check it.

Reply:
I think both `Instant` and `Pair<LocalDateTime, ZoneOffset>` are suitable, but yeah, the first should be simpler to implement. The only potential downside with `Instant` I see is that in a round-trip server-client-server, the time zone information is lost.

`kotlinx.datetime.Instant.parse("2020-07-31T09:16:15+02:00")` currently fails (as of `kotlinx.datetime` version 0.1.1), however, so we'd need custom parsing logic regardless.

Reply:
[The Java implementation ran into a similar conundrum](https://github.com/swagger-api/swagger-codegen/issues/2138#issuecomment-224254752)

Reply:
Yes the Zoning info probably requires us to use some offset stuff, I'm investigating writing a regex

Reply:
We can easily add a method to convert it to an instant, so the loss of information is optional

Reply:
[This is something I threw together quickly when trying to investigate this.](https://gist.github.com/justasm/d34c06411940643874eda051e72ce2e4)

Reply:
I'll also raise an issue with them

Reply:
> [This is something I threw together quickly when trying to investigate this.](https://gist.github.com/justasm/d34c06411940643874eda051e72ce2e4)

Seems well enough, although the last case should probably raise an error

Reply:
Not complete yet and missing some documentation, but a start I guess: https://gitlab.com/-/snippets/2056386
(not tested yet, feel free to test it)

Reply:
Fixed, please test if that works for you